### DISCUSSION TITLE
Is it possible to have ontime display the time as a string in Qlab for use on a Qlab attached screen?

### DISCUSSION THREAD

**Discussion by michaellammers** [cpvalente/ontime#1351] [Q&A] [answered]
I am not really OSC savvy, but managed to get some data out of Ontime to Qlab by enabling OSC input on port 53005 and output on port 53000 in Ontime and enabling Network Output on port 53005 in QLab. 
With an OSC integration set to cycle every second, address /cue/Einscherm/livetext and Arguments "{{timer.current}}" I have been able to get Ontime push the time to that Cue in QLab. But... it is in milliseconds and not in human readable time. 
1. Is the above the proper way to use Ontimes' countdown in QLab?
2. Can I get the time in HH:MM:ss format?

Thanks in advance!
Michaël

![Scherm­afbeelding 2024-11-30 om 15 30 08](https://github.com/user-attachments/assets/08f83c99-d868-488e-b9fd-8f4d76ded184)

**Comment by cpvalente** [comment, marked as answer]
Hi @michaellammers , thank you for taking the time to reach out

It seems like you figured most of it out but I can definitely help with your questions

> 1. Is the above the proper way to use Ontimes' countdown in QLab?

What you describe sounds reasonable to me, I am unsure if you need the circular feedback (port in and out) but the setup is sound.
I would maybe question whether you need to update `Every second`, it seems than `On Timer Update` might be better indicated to the data you are sending. But it depends on whether you want the empty timer in between cues

> 2. Can I get the time in HH:MM:ss format?

The templating language that you are using (the bit inside {{}}) has a few human readable values. You can read more about it in the docs at https://docs.getontime.no/api/integrations/#human-readable-data

Meanwhile, what you are looking for is likely
`{{human.runningTimer}}`

Hope this helps, feel free to follow up otherwise
You can also hop in to our [discord server](https://discord.com/invite/eje3CSUEXm) and you can DM me there

**Comment by michaellammers** [comment]
Great feedback, highly appreciated. I tested and it also works without 'input'. The templating is absolutely great, I've checked the docs thoroughly, but as an OSC newbe I sometimes did not know where to look. Thanks for putting me in the right direction!
For your info: I plan to use the timer in an escaperoom. Thanks for this wonderful tool!
